### PR TITLE
Move plugins conflict detection helper

### LINKED ISSUE #2341: Move plugins conflict detection helper
## Type of report

Task

## Provide description of the task

Currently this helper is in `CKEDITOR.tools`, but it would look better in `editor.plugins` namespace. Let's move it!

## Other details

* Browser: …
* OS: …
* CKEditor version: 4.10.1 (`master`)
* Installed CKEditor plugins: …

Followup #2249.

### PR DESCRIPTION
**Pull request by Comandeer**
## What is the purpose of this pull request?

Task

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?

Moved `detectPluginConflict` from `CKEDITOR.tools` to `CKEDITOR.editor.prototype.plugins`. Updated tests accordingly.

Closes #2341.

### CODE REVIEW

File: core/editor.js [hunk 1]
```diff
@@ -524,21 +524,7 @@
 			// The list of URLs to language files.
 			var languageFiles = [];
 
-			/**
-			 * An object that contains references to all plugins used by this
```
Comment (jacekbogdanski) [inline core/editor.js]: `editor.plugins` object docs are not visible with umberto built - not sure if they should, would like to know your opinion.

**Comment by jacekbogdanski** [comment]
I'm also not sure about naming, `plugins.detectConflicts` sounds better for me than `plugins.detectPluginsConflict` because we are not repeating property name.

**Comment by Comandeer** [comment]
Good catch, I've updated name to `detectConflict`.

File: core/editor.js (hunk 1 above)
Comment (Comandeer) [reply core/editor.js]: Hmm, they appear as a description of `plugins` property in `editor` page. Seems like Umberto upstream issue.

**Comment by Comandeer** [comment]
Added docs changes suggested by @mlewand.

**Pull request merged**